src/js/plugins/ajax.js

```diff
@@ -4,12 +4,30 @@ export default {
   /* istanbul ignore next */
   install(Vue, { store }) {
     /* istanbul ignore next */
-    Vue.prototype.$post = function(url, data = {}, opt) {
+    Vue.prototype.$post = function(url, data = {}, opt = {}) {
       // Set SESSION_ID if available
       if (!data.SESSION_ID && store.getters.sessionID) {
         data.SESSION_ID = store.getters.sessionID
       }
 
+      /**
+       * We must add the scroll_version_id from the request back
+       * into the response.  This is necessary for tracking
+       * ID's, since we use the meta ID's like col_id and
+       * artefact_id to access data in the col_data and
+       * artefact_shape tables, and thus would not get the
+       * proper entry without knowing the relevant scroll_version_id.
+       * We simply insert the request back in to the response.
+       */
+      if (data.scroll_version_id) {
```

Comment:
The upshot of all of this is that we expose the payload that we send to in the response. Is tha tright?

Reply:
Exactly!